### PR TITLE
Improve perf on CollectResults

### PR DESCRIPTION
**Pull request by forki**
There are situations where we can cut the name resolution

### CODE REVIEW

File: src/fsharp/NameResolution.fs [hunk 1]
```diff
@@ -2193,7 +2199,7 @@ let rec ResolveExprLongIdentPrim sink (ncenv:NameResolver) fullyQualified m ad n
               if List.isEmpty tcrefs then NoResultsOrUsefulErrors else
               let tcrefs = tcrefs |> List.map (fun tcref -> (resInfo,tcref))
               let tcrefs = CheckForTypeLegitimacyAndMultipleGenericTypeAmbiguities (tcrefs, TypeNameResolutionInfo.ResolveToTypeRefs (TypeNameResolutionStaticArgsInfo.Indefinite), PermitDirectReferenceToGeneratedType.No, unionRanges m id.idRange)
-              ResolveLongIdentInTyconRefs ncenv nenv LookupKind.Expr 1 m ad rest typeNameResInfo id.idRange tcrefs
+              ResolveLongIdentInTyconRefs true ncenv nenv LookupKind.Expr 1 m ad rest typeNameResInfo id.idRange tcrefs
```
Comment (KevinRansom) [inline src/fsharp/NameResolution.fs]: Can we use an enum for at most 1, true is a tad tricky to read?

File: src/fsharp/NameResolution.fs [hunk 2]
```diff
@@ -2525,13 +2531,14 @@ let rec ResolveTypeLongIdentPrim (ncenv:NameResolver) fullyQualified m nenv ad (
...
+
         let modulSearch = 
-            ResolveLongIndentAsModuleOrNamespaceThen ncenv.amap m fullyQualified nenv ad lid 
+            ResolveLongIndentAsModuleOrNamespaceThen false ncenv.amap m fullyQualified nenv ad lid 
                 (ResolveTypeLongIdentInModuleOrNamespace ncenv typeNameResInfo ad genOk)
             |?> List.concat 
 
         let modulSearchFailed() = 
-            ResolveLongIndentAsModuleOrNamespaceThen ncenv.amap m fullyQualified nenv AccessibleFromSomeFSharpCode lid 
+            ResolveLongIndentAsModuleOrNamespaceThen false ncenv.amap m fullyQualified nenv AccessibleFromSomeFSharpCode lid
```
Comment (KevinRansom) [inline src/fsharp/NameResolution.fs]: Ditto enum for false.

**Comment by KevinRansom** [comment]
@dotnet-bot test this.

failures are not related.

File: src/fsharp/NameResolution.fs (hunk 1 above)
Comment (forki) [inline src/fsharp/NameResolution.fs]: Yes would have done this anyways. But you are definitely right

**Comment by forki** [comment]
I'm trying to measure this and it's not making huge impact as is on a full compile. But from what I can in VS debug it's definitely doing a lot less work for name resolution. This is especially important for suggestion errors.

**Comment by forki** [comment]
made it an emu

**Comment by forki** [comment]
Wow didn't expect this to be merged so fast. @dsyme could you please take a
look as well?

Am 15.12.2016 9:43 nachm. schrieb "Kevin Ransom (msft)" <
notifications@github.com>:

Merged #2018 <https://github.com/Microsoft/visualfsharp/pull/2018>.

—
You are receiving this because you authored the thread.
Reply to this email directly, view it on GitHub
<https://github.com/Microsoft/visualfsharp/pull/2018#event-895831905>, or mute
the thread

**Pull request merged**